packages/controller/lib/setup/setupUpload.js

```diff
@@ -464,343 +421,345 @@ function Upload(options) {
-                iopack.common = iopack.common || {};
-                iopack.common.installedFrom = iopackFile.common.installedFrom;
+            if (ioPackFile && ioPackFile.common && ioPackFile.common.installedFrom) {
+                ioPack.common = ioPack.common || {};
+                ioPack.common.installedFrom = ioPackFile.common.installedFrom;
+            }
+            // Not existing? Why ever ... we recreate
+            let obj = await objects.getObject('system.adapter.' + name);
+            obj = obj || {};
+            obj.common = ioPack.common || {};
+            obj.native = ioPack.native || {};
+            // protected/encryptedNative and notifications also need to be updated
+            obj.protectedNative = ioPack.protectedNative || [];
+            obj.encryptedNative = ioPack.encryptedNative || [];
+            obj.notifications = ioPack.notifications || [];
+            // update instanceObjects and objects
+            obj.instanceObjects = ioPack.instanceObjects || [];
+            obj.objects = ioPack.objects || [];
+
+            obj.type   = 'adapter';
+
+            obj.common.installedVersion = ioPack.common.version;
+
+            if (obj.common.news) {
+                delete obj.common.news; // remove this information as it could be big, but it will be taken from repo
             }
-            objects.getObject(`system.adapter.${name}`, (err, obj) => {
-                if (err || !obj) {
-                    // Not existing? Why ever ... we recreate
-                    obj = {};
-                }
-                obj.common = iopack.common || {};
-                obj.native = iopack.native || {};
-                // protected/encryptedNative and notifications also need to be updated
-                obj.protectedNative = iopack.protectedNative || [];
-                obj.encryptedNative = iopack.encryptedNative || [];
-                obj.notifications = iopack.notifications || [];
-                // update instanceObjects and objects
-                obj.instanceObjects = iopack.instanceObjects || [];
-                obj.objects = iopack.objects || [];
-
-                obj.type   = 'adapter';
-
-                obj.common.installedVersion = iopack.common.version;
-
-                if (obj.common.news) {
-                    delete obj.common.news; // remove this information as it could be big, but it will be taken from repo
-                }
 
-                const hostname = tools.getHostName();
+            const hostname = tools.getHostName();
 
-                obj.from = `system.host.${hostname}.cli`;
-                obj.ts = Date.now();
+            obj.from = `system.host.${hostname}.cli`;
+            obj.ts = Date.now();
 
-                objects.setObject('system.adapter.' + name, obj, () =>
-                    this._upgradeAdapterObjectsHelper(name, iopack, hostname, logger, callback));
-            });
+            await objects.setObjectAsync('system.adapter.' + name, obj);
```

Comment:
error here was ignored in old code, now would prevent the  next call

Reply:
Added try/catch